### PR TITLE
Add screenshot crop and mask functionality to library.

### PR DESCRIPTION
**Pull request by marcrent**
Added the optional `crop` and `masks` arguments to `screenshot_matches` assertion.

### CODE REVIEW

File: docs/usage.rst [hunk 1]
```diff
@@ -5,3 +5,15 @@ Usage
 To use pytest-selenium-pdiff in a project::
 
     import pytest-selenium-pdiff
+
+    def test_example(self, driver):
+        do_something()
+        screenshot_matches(driver, 'module/my screen shot', crop=(0,0,200,200), masks=[(20,30,50,70)]
```
Comment (pplante) [inline docs/usage.rst]: Mind altering the example so it covers all 3 variations separately?  Right now it looks like all these arguments are required.

File: pytest_selenium_pdiff/assertions.py [hunk 2]
```diff
@@ -26,21 +38,27 @@ def screenshot_matches(driver, screenshot_name):
 
     driver.get_screenshot_as_file(captured_screenshot)
 
+    try:
+        transform_screen_shot(captured_screenshot, transformed_screenshot, crop=crop, masks=masks)
+    except Exception:
```
Comment (fearofcode) [inline pytest_selenium_pdiff/assertions.py]: any way to make this less broad?

File: pytest_selenium_pdiff/assertions.py [hunk 3]
```diff
@@ -4,9 +4,20 @@
...
+    image = Image.open(original)
+    if crop:
+        image = image.crop(crop)
+    draw = ImageDraw.Draw(image)
+    for mask in masks:
+        draw.rectangle(mask, fill='magenta')
+    image.save(result)
+
+
+def screenshot_matches(driver, screenshot_name, crop=None, masks=[]):
```
Comment (pplante) [inline pytest_selenium_pdiff/assertions.py]: `masks` should be None in the function signature.  You can default it in the function body if you would like.

File: pytest_selenium_pdiff/assertions.py [hunk 4]
```diff
@@ -26,21 +38,27 @@ def screenshot_matches(driver, screenshot_name):
 
     driver.get_screenshot_as_file(captured_screenshot)
 
+    try:
+        transform_screen_shot(captured_screenshot, transformed_screenshot, crop=crop, masks=masks)
+    except Exception:
+        raise exceptions.InvalidCropOrMasks(screenshot_name, crop, masks)
```
Comment (pplante) [inline pytest_selenium_pdiff/assertions.py]: Could we provide better feedback than this message?  Possibly validate that a mask is a set of 4 ints?

File: requirements_dev.txt [hunk 5]
```diff
@@ -7,11 +7,12 @@ coverage==4.0
 Sphinx==1.3.1
 cryptography==1.3.2
 PyYAML==3.11
-pytest==3.0.1
+pytest==3.0.7
```
Comment (pplante) [inline requirements_dev.txt]: You can restore pytest to the original version.

File: tests/fixtures/page1-bold.html [hunk 6]
```diff
@@ -0,0 +1,20 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Title</title>
+    <style>
+        @import url('https://fonts.googleapis.com/css?family=Ubuntu:300,400');
```
Comment (pplante) [inline tests/fixtures/page1-bold.html]: External resources might cause sporadic test failures if the network is unexpectedly slow.  I am not entirely sure why you even added this font.

File: pytest_selenium_pdiff/assertions.py (hunk 2 above)
Comment (marcrent) [inline pytest_selenium_pdiff/assertions.py]: I removed this in favor of better argument checking.

File: pytest_selenium_pdiff/assertions.py (hunk 4 above)
Comment (marcrent) [inline pytest_selenium_pdiff/assertions.py]: Added better argument checking.

File: requirements_dev.txt (hunk 5 above)
Comment (marcrent) [inline requirements_dev.txt]: Circle CI breaks for some reason when I use this. I am inclined to leave it at 7.

File: docs/usage.rst (hunk 1 above)
Comment (marcrent) [inline docs/usage.rst]: Done.

**Pull request merged**